File: 110-balanced-binary-tree/110-balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    if (root === null) return true
    
    return getHeight(root) !== -1
};
var getHeight = (root) =>{
    if (root === null) return 0
    
    let left  = getHeight(root.left)
    let right = getHeight(root.right)
    
    if (left === -1 || right === -1 || Math.abs(left-right) > 1){
        return -1
    }
    return Math.max(left, right) + 1
}

File: 110-balanced-binary-tree/110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isBalanced(self, root):
        def dfs(root):
            if not root:
                return [True, 0]
        
            left_height  = dfs(root.left)
            right_height = dfs(root.right)
            is_balanced   = (left_height[0] and right_height[0] and
                             abs(left_height[1] - right_height[1]) <= 1)
        
            return [is_balanced, max(left_height[1], right_height[1]) + 1]
        
        return dfs(root)[0]
    
    
            
           